### PR TITLE
Make migrations scripts

### PR DESCRIPTION
**Pull request by haneslinger**
Cute little script to modified BuildingSync files from 2.4 to 2.5

### CODE REVIEW

File: migration_scripts/README.md
Comment (ThibaultMarzullo) [inline migration_scripts/README.md:1]: Can you add an example of usage please?

File: migration_scripts/upgrade_from_2.4_to_2.5.py
Comment (ThibaultMarzullo) [inline migration_scripts/upgrade_from_2.4_to_2.5.py:1]: Looks good!

File: migration_scripts/upgrade_from_2.4_to_2.5.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from lxml import etree as ET
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("file_path", type=str, help="path to file to update")
+    return parser.parse_args()
+
+def update_usefulLife(tree):
+    for usefulLife in tree.findall('.//*{http://buildingsync.net/schemas/bedes-auc/2019}UsefulLife'):
+        usefulLife.text = usefulLife.text.split(".", 1)[0]
```
Comment (ThibaultMarzullo) [inline migration_scripts/upgrade_from_2.4_to_2.5.py:14]:
```suggestion
        usefulLife.text = str(int(usefulLife.text))
```
Currently, we only round down, is this right? If so, would this also round up when necessary?

File: migration_scripts/upgrade_from_2.4_to_2.5.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    file_path = get_args().file_path
+
+    try:
+        tree = ET.parse(file_path)
+    except (FileNotFoundError, ET.ParseError) as e:
+        exit(f"File could not be read \n{e} \naborting...")
+
+    update_usefulLife(tree)
+    
+    tree.write(file_path)
```
Comment (ThibaultMarzullo) [inline migration_scripts/upgrade_from_2.4_to_2.5.py:26]: Can you save the original file under a different name like `ye_olde_excemelle` or a sober and more appropriate suffix like `my_xml_old`?

**Review by ThibaultMarzullo — CHANGES_REQUESTED** [review]
This looks good, thank you! On top of the comments that I made throughout the files, can you also include one file for testing? If we do not have an example XML with the right fields, maybe you can amend a minimalist XML like [this L000 pre-simulation XML](https://github.com/BuildingSync/use-cases/blob/main/BuildingSync-Gem/examples/L000_Pre-Simulation-1.0.0_01.xml).

**Comment by haneslinger** [comment]
> This looks good, thank you! On top of the comments that I made throughout the files, can you also include one file for testing? If we do not have an example XML with the right fields, maybe you can amend a minimalist XML like [this L000 pre-simulation XML](https://github.com/BuildingSync/use-cases/blob/main/BuildingSync-Gem/examples/L000_Pre-Simulation-1.0.0_01.xml).

There are mutliple v2.4 files in the repo currently, for example, `docs/notebooks/bsync_examples/example-smalloffice-level2.xml` (UsefulLife on L1408). Are you looking for something more specific?

**Comment by JieXiong9119** [comment]
This one looks good, but I want to point out that we had something built for version converting by using xsl in [here](https://github.com/BuildingSync/schema/tree/develop-v3/translation). I'm not sure if xsl is something more generic for xml conversion and might be useful for future development.

**Comment by haneslinger** [comment]
oh, this would have been great to know about. I haven't worked with xsl, let me dig into it.

**Comment by haneslinger** [comment]
wait, these scripts are for v2 to v3, what happened to v3?

**Comment by JieXiong9119** [comment]
> wait, these scripts are for v2 to v3, what happened to v3?

v3 is a frozen version that was determined not to be released in the near future. I don't think you want to dig into xslt - I suggest we go with pure Python but refer to that v3 xslt tool for the extensibility of this script (e.g. adding a comment note like the one in v3 converter such as "This BuildingSync v2.5 document was generated from a BuildingSync v2.4 document via ....").

**Review by ThibaultMarzullo — APPROVED** [review]
This is looking good, thanks Hannah!
I like Jie's suggestion of adding a comment saying that an XML was converted from 2.4 to 2.5 with this script.

**Comment by haneslinger** [comment]
I want to be clear: You want a comment in the generated/updated xml file stating it was modified by the script?

**Comment by JieXiong9119** [comment]
> I want to be clear: You want a comment in the generated/updated xml file stating it was modified by the script?

Yes. Something like [this line](https://github.com/BuildingSync/schema/blob/7edc2f9a5a57041ffa164b16be398d9f0917346c/translation/v2_to_v3.xsl#L21C22-L21C22) in the xslt tool for v2-to-v3, which adds a comment on top of the generated xml.

**Review by JieXiong9119 — APPROVED** [review]
Looks good. Thank you Hannah!

**Pull request merged**